Sources/Danger/GitLabDSL.swift

```diff
@@ -0,0 +1,222 @@
+import Cocoa
+
+public struct GitLab: Decodable {
+    public enum CodingKeys: String, CodingKey {
+        case mergeRequest = "mr"
+        case metadata
+    }
+
+    public let mergeRequest: GitLabMergeRequest
+    public let metadata: GitLabMetadata
+}
+
+public struct GitLabMetadata: Decodable, Equatable {
+    public let pullRequestID: String
+    public let repoSlug: String
+}
+
+public struct GitLabMergeRequest: Decodable, Equatable {
+    public enum State: String, Decodable {
+        case closed
+        case locked
+        case merged
+        case open
+    }
+
+    public struct Milestone: Decodable, Equatable {
+        public enum CodingKeys: String, CodingKey {
+            case createdAt = "created_at"
+            case description
+            case dueDate = "due_date"
+            case id
+            case iid
+            case projectId = "project_id"
+            case startDate = "start_date"
+            case state
+            case title
+            case updatedAt = "updated_at"
+            case webUrl = "web_url"
+        }
+
+        public enum State: String, Decodable {
```

Comment:
I don't know all the states for the models with one item only, still trying to figure out what they are